src/Uno.UI/Extensions/NSDateExtensions.iOS.cs

```diff
@@ -32,10 +32,10 @@ public static NSDate ToNSDate(this TimeSpan time)
 			return DateTime.Today.Add(time).ToNSDate();
 		}
 
-		internal static TimeSpan ToTimeSpan(this NSDate date, nint offsetInSecondsFromGMT)
+		internal static TimeSpan ToTimeSpanOfDay(this NSDate date, nint offsetInSecondsFromGMT)
 		{
 			var offset = TimeSpan.FromSeconds(offsetInSecondsFromGMT);
-			return date.ToTimeSpan().Add(offset);
+			return date.ToTimeSpan().Add(offset).NormalizeToDay();
```

Comment:
Please add runtime tests to validate that behavior. It seems that it may be easy to break or get wrong.

You can add one here: https://github.com/unoplatform/uno/tree/master/src/Uno.UI.RuntimeTests/Tests

Reply:
Or if the test is not dependent on the  platform, create a test here: https://github.com/unoplatform/uno/tree/master/src/Uno.UI.Tests

Reply:
Thank you, I will add tests.